### ISSUE TITLE
Compiling in Alpine linux

### ISSUE THREAD

**Issue by MorganPeterson** [deadpixi/mtm#43]
Compiling in Alpine linux from the master branch results in the compiler not finding curses.h and not having the man directory /usr/local.

I made these changes in config.def.h:

`diff --git a/config.def.h b/config.def.h
index 248b296..f6ed0fc 100644
--- a/config.def.h
+++ b/config.def.h
@@ -61,7 +61,7 @@
 
 /* The path for the wide-character curses library. */
 #ifndef NCURSESW_INCLUDE_H
-    #if defined(__APPLE__) || !defined(__linux__) || defined(__FreeBSD__)
+    #if defined(__APPLE__) || !defined(__linux__) || !defined(linux) || defined(__FreeBSD__)
         #define NCURSESW_INCLUDE_H <curses.h>
     #else
         #define NCURSESW_INCLUDE_H <ncursesw/curses.h>
`
Then compiled it this way:
`MAKEDIR=/usr/share make && make install`

That allowed it to compile.

**Comment by GuillaumeFavelier** [comment]
Thanks for reporting this!

I was able to compile it on my Arch linux once I applied the patch you described.

I just typed `y` when:
```
cp -i config.def.h config.h
cp: overwrite 'config.h'?
```

**Comment by domrc** [comment]
I do not get what this patch does for Arch Linux. `mtm` compiles fine, without modification, from https://aur.archlinux.org/packages/mtm-git/ (for me at least, on multiple x86_64 machines)
I can not speak for Alpine Linux.

**Comment by JackalSix** [comment]
Well if you look at the pkgbuild on aur it runs make with:
`HEADERS='-DNCURSESW_INCLUDE_H="<ncurses.h>"'`
So it looks like they just fix the same thing this patch would manually.

**Comment by domrc** [comment]
I quick decided to see if the patch would remove the need for said `$HEADERS` variable. I modified the `PKGBUILD`  to comment out the `HEADERS=...` from the `make` line.  I checked to see if it would build without it. It did not compile.
Then I used a `prepare()` function to patch the `config.h` and add the `|| !defined(linux)` to the line modified in the patch. It didn't either.

**TL;DR** @JackalSix, I tried to see if that was the case, but it doesn't seem that way. Or maybe I'm missing something.

P.S. I then tried patching in `|| defined(linux)` (not inverting this time) and got a compile without that `HEADERS` variable. *But* I don't know if that will mess with compiling on other Linux distros. `HEADERS` just seems safer at the moment.
